syft/frameworks/torch/crypto/securenn.py

```diff
@@ -284,30 +293,24 @@ def share_convert(a_sh):
     # 2)
     a_tilde_sh = a_sh + r_sh
     a_shares = a_sh.child
-    ptr0 = a_shares[workers[0].id] + r_shares[workers[0].id]
-    beta0 = ((a_shares[workers[0].id] + r_shares[workers[0].id]) >= L).long() - (
-        (a_shares[workers[0].id] + r_shares[workers[0].id]) < 0
-    ).long()
-    ptr1 = a_shares[workers[1].id] + r_shares[workers[1].id]
-    beta1 = ((a_shares[workers[1].id] + r_shares[workers[1].id]) >= L).long() - (
-        (a_shares[workers[1].id] + r_shares[workers[1].id]) < 0
-    ).long()
+    beta0 = ((a_shares[workers[0].id] + r_shares[workers[0].id]) >= L).long()
+    beta1 = ((a_shares[workers[1].id] + r_shares[workers[1].id]) >= L).long()
     beta = sy.MultiPointerTensor(children=[beta0.long(), beta1.long()])
 
     # 4)
     a_tilde_shares = a_tilde_sh.child
     delta = (
         ((a_tilde_shares[workers[0].id] * 1).get() + (a_tilde_shares[workers[1].id] * 1).get()) >= L
     ).long()
-    x = a_tilde_sh.get()
+    x = a_tilde_sh.get() % L
 
     # 5)
     x_bit = decompose(x)
     x_bit_sh = x_bit.share(*workers, field=p, crypto_provider=crypto_provider).child
     delta_sh = delta.share(*workers, field=L - 1, crypto_provider=crypto_provider).child
 
     # 6)
-    eta_p = private_compare(x_bit_sh, r, eta_pp)
+    eta_p = private_compare(x_bit_sh, r - 1, eta_pp)
```

Comment:
You should keep `r` (ref: Algo 4 line 6)

Reply:
Are you sure? I see `r-1` written...

Reply:
In have `r` in https://www.microsoft.com/en-us/research/uploads/prod/2018/09/securenneprint.pdf
But https://eprint.iacr.org/2018/442.pdf says `r-1` haha that's right

So... let's see. I believe you're right `r-1` should be used. The eprint has been updated in 08-Mar-2019 and `r-1` has been put instead of `r`. And it seems to make sense from the math point of view